include/flatbuffers/idl.h

```diff
@@ -47,24 +47,24 @@ namespace flatbuffers {
 // of type tokens.
 // clang-format off
 #define FLATBUFFERS_GEN_TYPES_SCALAR(TD) \
-  TD(NONE,   "",       uint8_t,  byte,   byte,    byte,   uint8,   u8) \
-  TD(UTYPE,  "",       uint8_t,  byte,   byte,    byte,   uint8,   u8) /* begin scalar/int */ \
-  TD(BOOL,   "bool",   uint8_t,  boolean,bool,    bool,   bool,    bool) \
-  TD(CHAR,   "byte",   int8_t,   byte,   int8,    sbyte,  int8,    i8) \
-  TD(UCHAR,  "ubyte",  uint8_t,  byte,   byte,    byte,   uint8,   u8) \
-  TD(SHORT,  "short",  int16_t,  short,  int16,   short,  int16,   i16) \
-  TD(USHORT, "ushort", uint16_t, short,  uint16,  ushort, uint16,  u16) \
-  TD(INT,    "int",    int32_t,  int,    int32,   int,    int32,   i32) \
-  TD(UINT,   "uint",   uint32_t, int,    uint32,  uint,   uint32,  u32) \
-  TD(LONG,   "long",   int64_t,  long,   int64,   long,   int64,   i64) \
-  TD(ULONG,  "ulong",  uint64_t, long,   uint64,  ulong,  uint64,  u64) /* end int */ \
-  TD(FLOAT,  "float",  float,    float,  float32, float,  float32, f32) /* begin float */ \
-  TD(DOUBLE, "double", double,   double, float64, double, float64, f64) /* end float/scalar */
+  TD(NONE,   "",       uint8_t,  byte,   byte,    byte,   uint8,   u8,   Byte) \
+  TD(UTYPE,  "",       uint8_t,  byte,   byte,    byte,   uint8,   u8,   Byte) /* begin scalar/int */ \
+  TD(BOOL,   "bool",   uint8_t,  boolean,bool,    bool,   bool,    bool, Boolean) \
+  TD(CHAR,   "byte",   int8_t,   byte,   int8,    sbyte,  int8,    i8,   Byte) \
+  TD(UCHAR,  "ubyte",  uint8_t,  byte,   byte,    byte,   uint8,   u8,   Byte) \
```

Comment:
Should use `UByte` instead of `Byte`. Similar for other unsigned number types.

Reply:
Hi @LouisCAD, the idea was to have the initial implementation as close as possible to the java one, to make the MR simpler, easier to compare generate code with java counterpart and use the same tests as the java version. Late on started to make it more idiomatic and adding kotlin specific features in a more self-contained, easy to review way.

So in my opinion we should keep the types in this MR the same as used in java. Also AFAIK unsigned it only available in latest version of Kotlin(1.3) and still experimental(https://kotlinlang.org/docs/reference/basic-types.html#experimental-status-of-unsigned-integers)

Reply:
When (if?) unsigned integers become no longer experimental, are we going to get an API breakage? Or need a flag to allow to opt-in/out to/from always signed mode?

API breakages make people defer upgrades, which can cause long term issues, the flag/option migh be closer to a desirable option, even if first release can support unsigned integers since it is indeed experimental and might break too.

Reply:
Very much agree with @LouisCAD that unsigned types should be part of the initial PR. I wasn't aware that Kotlin had them. The Java implementation has a bunch of ugly hacks (using bigger types e.g. `long` to represent `uint`) which would be great to remove.

In general, while I understand "stay close to Java" as a reasonable goal, anything that would generate huge API breakage later should really be done right directly. It will save so much trouble later. While in a new API some breakage is acceptable and unavoidable, the more we can get right in the first version will really simplify things.

Reply:
Alright folks, I will mark this as unresolved and take a look a the experimental unsigned feature.

Reply:
That also means I will not be able to use  java's FlatBufferBuilder.java, since it does not support unsigned out of the box. I would have to create own Kotlin runtime library.

Reply:
If you create own Kotlin runtime library, then you can remove dependencies on the JVM and make it multiplatform straightaway.

Reply:
Alright, I am now using the experimental unsigned API, as suggested. 

Keep in mind  that this API is still experimental and marked as "Moving Fast".

https://kotlinlang.org/docs/reference/evolution/components-stability.html

But the usage of bigger type for unsigned type was indeed removed.